File: go-cli/internal/handler/list.go
// Package handler contains CLI command handlers.
package handler

import (
	"fmt"

	"convex-notes-cli/internal/service"
	"convex-notes-cli/internal/ui"

	"github.com/fatih/color"
)

// ListHandler handles the list notes command.
type ListHandler struct {
	service *service.NoteService
}

// NewListHandler creates a new ListHandler.
func NewListHandler(svc *service.NoteService) *ListHandler {
	return &ListHandler{service: svc}
}

// Execute runs the handler.
func (h *ListHandler) Execute() error {
	blue := color.New(color.FgBlue)
	green := color.New(color.FgGreen)
	yellow := color.New(color.FgYellow)

	blue.Println("\n📋 Fetching notes...")

	notes, err := h.service.List()
	if err != nil {
		return fmt.Errorf("failed to list notes: %w", err)
	}

	if len(notes) == 0 {
		yellow.Println("\n📭 No notes found. Create your first one!")
		return nil
	}

	green.Printf("\nFound %d note(s)\n\n", len(notes))
	ui.DisplayNotes(notes)

	return nil
}

File: zig-cli/src/handler/list.zig
//! List notes handler.

const std = @import("std");

const service = @import("../service/note.zig");
const NoteService = service.NoteService;
const table = @import("../ui/table.zig");

/// Execute list command.
pub fn execute(svc: *NoteService) void {
    std.debug.print("\x1b[34m\n📋 Fetching notes...\x1b[0m\n", .{});

    const notes = svc.list() catch |err| {
        std.debug.print("\x1b[31m❌ Failed to list notes: {}\x1b[0m\n", .{err});
        return;
    };

    if (notes.len == 0) {
        std.debug.print("\x1b[33m\n📭 No notes found. Create your first one!\x1b[0m\n", .{});
    } else {
        std.debug.print("\x1b[32m\nFound {d} note(s)\x1b[0m\n", .{notes.len});
        table.display(notes);
    }
}
